lifecycle_test.go

```diff
@@ -456,6 +461,166 @@ func deleteReleaseBundleWithProject(t *testing.T, lcManager *lifecycle.Lifecycle
 	time.Sleep(5 * time.Second)
 }
 
+func TestSetReleaseBundleTag(t *testing.T) {
+
+	cleanCallback := initLifecycleTest(t, artifactoryLifecycleSetTagMinVersion)
+	defer cleanCallback()
+	lcManager := getLcServiceManager(t)
+
+	deleteBuilds := uploadBuilds(t)
+	defer deleteBuilds()
+
+	//set tag
+	createRbFromSpec(t, tests.LifecycleBuilds12, tests.LcRbName1, number1, true, true)
+	defer deleteReleaseBundle(t, lcManager, tests.LcRbName1, number1)
+	assertStatusCompleted(t, lcManager, tests.LcRbName1, number1, "")
+	defer setReleaseBundleTag(t, lcManager, tests.LcRbName1, number1, "", "bundle-tag")
+
+	//unset tag
+	createRbFromSpec(t, tests.LifecycleBuilds12, tests.LcRbName2, number1, true, true)
+	defer deleteReleaseBundle(t, lcManager, tests.LcRbName2, number1)
+	assertStatusCompleted(t, lcManager, tests.LcRbName2, number1, "")
+	defer setReleaseBundleTag(t, lcManager, tests.LcRbName1, number1, "", "")
+}
+
+func TestReleaseBundleCreateOrUpdateProperties(t *testing.T) {
+
+	cleanCallback := initLifecycleTest(t, artifactoryLifecycleSetTagMinVersion)
+	defer cleanCallback()
+	lcManager := getLcServiceManager(t)
+
+	deleteBuilds := uploadBuilds(t)
+	defer deleteBuilds()
+
+	//set properties
+	createRbFromSpec(t, tests.LifecycleBuilds12, tests.LcRbName1, number1, true, true)
+	defer deleteReleaseBundle(t, lcManager, tests.LcRbName1, number1)
+	assertStatusCompleted(t, lcManager, tests.LcRbName1, number1, "")
+	defer setReleaseBundleProperties(t, lcManager, tests.LcRbName1, number1, "default",
+		"key1=value1;key2=value2")
+	defer setReleaseBundleProperties(t, lcManager, tests.LcRbName1, number1, "default",
+		"key1=value1;key2=''")
+
+}
+
+func TestReleaseBundleDeleteProperties(t *testing.T) {
+
+	cleanCallback := initLifecycleTest(t, artifactoryLifecycleSetTagMinVersion)
+	defer cleanCallback()
+	lcManager := getLcServiceManager(t)
+
+	deleteBuilds := uploadBuilds(t)
+	defer deleteBuilds()
+
+	//set properties
+	createRbFromSpec(t, tests.LifecycleBuilds12, tests.LcRbName1, number1, true, true)
+	defer deleteReleaseBundle(t, lcManager, tests.LcRbName1, number1)
+	assertStatusCompleted(t, lcManager, tests.LcRbName1, number1, "")
+	defer setReleaseBundleProperties(t, lcManager, tests.LcRbName1, number1, "default",
+		"key1=value1;key2=value2")
+	defer deleteReleaseBundleProperties(t, lcManager, tests.LcRbName1, number1, "default", "key1,key2")
```

Comment:
is defer needed here? since its the end instruction for the function it is redundant.